File: core/src/zyzxdev/swtch/util/FontLayoutManager.java
package zyzxdev.swtch.util;

import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.GlyphLayout;

import java.util.HashMap;

/**
 * Created by aaron on 5/29/17.
 */

public class FontLayoutManager{
	private static HashMap<StringAndFont, GlyphLayout> map = new HashMap<StringAndFont, GlyphLayout>();

	public static GlyphLayout getFontLayout(String str, BitmapFont font){
		StringAndFont saf = new StringAndFont(str, font);
		if(!map.containsKey(saf)) {
			System.out.println("NEW GLYPHLAYOUT FOR STR "+str);
			GlyphLayout l = new GlyphLayout();
			l.setText(font, str);
			map.put(saf, l);
			return l;
		}
		return map.get(saf);
	}

	private static class StringAndFont{
		private String str;
		private BitmapFont font;

		private StringAndFont(String str, BitmapFont font){
			this.str = str;
			this.font = font;
		}

		@Override
		public int hashCode(){
			return str.hashCode()*font.hashCode();
		}

		public boolean equals(Object o){
			return o instanceof StringAndFont && ((StringAndFont) o).font == font && ((StringAndFont) o).str.equals(str);
		}
	}
}

File: core/src/qixl/swtch/util/FontLayoutManager.kt
package qixl.swtch.util

import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.badlogic.gdx.graphics.g2d.GlyphLayout

import java.util.HashMap

/**
 * Created by aaron on 5/29/17.
 */

object FontLayoutManager {
    private val map = HashMap<StringAndFont, GlyphLayout>()

    fun getFontLayout(str: String, font: BitmapFont): GlyphLayout? {
        val saf = StringAndFont(str, font)
        if (!map.containsKey(saf)) {
            println("NEW GLYPHLAYOUT FOR STR " + str)
            val l = GlyphLayout()
            l.setText(font, str)
            map.put(saf, l)
            return l
        }
        return map[saf]
    }

    private class StringAndFont constructor(private val str: String, private val font: BitmapFont) {

        override fun hashCode(): Int {
            return str.hashCode() * font.hashCode()
        }

        override fun equals(o: Any?): Boolean {
            return o is StringAndFont && o.font === font && o.str == str
        }
    }
}
